### ISSUE TITLE
Random notifications with "documentColumnIndex cannot be greater than the line's maximum index"

### ISSUE THREAD

**Issue by jemag** [Microsoft/vscode-azurearmtools#193]
Repro steps:
Really hard to tell what causes this specific issue as it seems fairly random.

I think it might happen when autocomplete suggestions are triggered. I use a smaller vscode window where it only takes half of my screen. My only guess is that perhaps there are some situations where the autocompletion menu might not have enough space to render?!

Although this notification happens fairly often, I have not been able to determine the exact steps that will reproduce it 100% of the time.

Action: provideCompletionItems
Error type: ERR_ASSERTION
Error Message: documentColumnIndex (27) cannot be greater than the line's maximum index (26)

Version: 0.5.0
OS: Arch linux
VS Code Version: 1.32.2

<details>
<summary>Call Stack</summary>

```
Function.fromDocumentLineAndColumnIndexes (dist/extension.bundle.js:127:96193)
DeploymentTemplate.getContextFromDocumentLineAndColumnIndexes (dist/extension.bundle.js:127:92640)
Object.<anonymous> (dist/extension.bundle.js:127:125835)
Generator.next (<anonymous>)
module.exports.n (dist/extension.bundle.js:127:117380)
new Promise (<anonymous>)
module.exports.n (dist/extension.bundle.js:127:117157)
Object.<anonymous> (dist/extension.bundle.js:127:125744)
Object.<anonymous> (dist/extension.bundle.js:1:182883)
Generator.next (<anonymous>)
module.exports.n (dist/extension.bundle.js:1:181258)
new Promise (<anonymous>)
module.exports.n (dist/extension.bundle.js:1:181035)
Object.module.exports.t.callWithTelemetryAndErrorHandling (dist/extension.bundle.js:1:182800)
AzureRMTools.<anonymous> (dist/extension.bundle.js:127:125667)
Generator.next (<anonymous>)
module.exports.n (dist/extension.bundle.js:127:117380)
new Promise (<anonymous>)
module.exports.n (dist/extension.bundle.js:127:117157)
AzureRMTools.onProvideCompletionItems (dist/extension.bundle.js:127:125562)
Object.provideCompletionItems (dist/extension.bundle.js:127:121876)
define.provideCompletionItems.a.asPromise (extensionHostProcess.js:569:293)
t.asPromise.Promise (extensionHostProcess.js:48:58)
new Promise (<anonymous>)
Object.t.asPromise (extensionHostProcess.js:48:32)
O.provideCompletionItems (extensionHostProcess.js:569:264)
define.$provideCompletionItems._withAdapter.e (extensionHostProcess.js:582:811)
$._withAdapter (extensionHostProcess.js:575:502)
$provideCompletionItems (extensionHostProcess.js:582:789)
h._doInvokeHandler (extensionHostProcess.js:593:118)
h._invokeHandler (extensionHostProcess.js:592:777)
h._receiveRequest (extensionHostProcess.js:591:383)
h._receiveOneMessage (extensionHostProcess.js:590:279)
define.constructor._protocol.onMessage.e (extensionHostProcess.js:588:565)
u.fire (extensionHostProcess.js:43:254)
e (extensionHostProcess.js:39:64)
u.fire (extensionHostProcess.js:43:254)
s (extensionHostProcess.js:168:754)
Socket.define.constructor._socketDataListener.e (extensionHostProcess.js:168:960)
Socket.emit (events.js:182:13)
addChunk (_stream_readable.js:279:12)
readableAddChunk (_stream_readable.js:264:11)
Socket.Readable.push (_stream_readable.js:219:10)
Pipe.onread (net.js:636:20)
```

</details>

**Comment by StephenWeatherford** [comment]
Thanks for reporting this.  I saw this for the first time yesterday.  I'll try to track it down.

**Comment by StephenWeatherford** [comment]
Haven't seen it again.

**Comment by StephenWeatherford** [comment]
Seems to be only happening since 0.5.0, maybe related to callWithTelemetryAndErrorHandling{,Sync} calls?  It appears the line/col are out of sync with the state of the deployment template.

For now, suppressing the error display.

**Comment by StephenWeatherford** [comment]
Caused by #221 which can happen while typing, which causes the parsed template to be out of sync with the sources, which causes this issue.

**Comment by StephenWeatherford** [comment]
FYI, this was fixed in the 0.6.0 release on Thursday.  Thanks again for reporting.
